.github/workflows/update-maintainers.yml

```
@@ -0,0 +1,245 @@
+name: Update maintainers list
+
+on:
+  push:
+    branches: ["main"]
+    paths: ["lib/maintainers.nix"]
```

I'd personally prefer:


```suggestion
    branches: main
    paths: lib/maintainers.nix
```

or


```suggestion
    branches:
      - main
    paths:
      - lib/maintainers.nix
```